model/host/db.go

```diff
@@ -781,6 +782,19 @@ func RemoveStaleInitializing(distroID string) error {
 		query[key] = distroID
 	}
 
+	hosts := []Host{}
+	if err := db.FindAll(Collection, query, db.NoProjection, db.NoSort, db.NoSkip, db.NoLimit, &hosts); err != nil {
+		return errors.WithStack(err)
+	}
+
+	catcher := grip.NewBasicCatcher()
+	for _, h := range hosts {
+		catcher.Add(h.DeleteJasperCredentials(ctx, env))
+	}
```

Comment:
rather than doing a DeleteOne in a loop can we change this to do a delete-many with a $in query?

Reply:
Done.